docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+# 1: IBC Architecture
+
+> This is an overview of the high-level architecture & dataflow of the IBC protocol.
+
+> For a broad set of protocol design principles, see [here](./2_IBC_DESIGN_PRINCIPLES.md).
+
+> For definitions of terms used in IBC specifications, see [here](./3_IBC_TERMINOLOGY.md).
+
+This document outlines the architecture of the authentication, transport, and ordering layers of the IBC protocol stack. This document does not describe specific protocol details — those are contained in individual ICSs.
+
+## What is IBC?
+
+The *inter-blockchain communication protocol* is designed for use as a reliable, authenticated module-to-module communication protocol between modules running on independent distributed ledgers across an untrusted network layer.
+
+## What is IBC not?
+
+IBC is not (only) an atomic-swap protocol: arbitrary cross-chain data transfer and computation is supported.
+
+IBC is not (only) a token transfer protocol: token transfer is a possible application-layer use of the IBC protocol.
+
+IBC is not (only) a sharding protocol: there is no single state machine being split across chains, but rather a diverse set of different state machines on different chains which share some common interfaces.
+
+IBC is not (only) a layer-two scaling protocol: all chains implementing IBC exist on the same "layer", although they may occupy different points in the network topology, and there is no single root chain or single validator set.
+
+## Motivation
```

Comment:
Good paragraph, but it takes the angle of scalability as primary motivation factor. I agree it is an important motivation factor, but isn't interoperability an other important motivation factor in itself (rather than a simple consequence of solving the scalability problem)? 

Even between two non-scalable ledgers interoperability is a strong desiderata. The ability for two ledgers to exchanges messages in a (somewhat) trustless way is a missing feature of the current blockchian ecosystem, and I think we should also mention it in this section. I would add a paragraph about interoperability before "The *interblockchain communication protocol* takes an orthogonal approach to a differently formulated version of the problem".

Reply:
I agree that interoperability is a desideratum, but I think it is derived from the primary goal of a scalable, versatile ledger or set of interacting ledgers - if a single ledger could handle a diversity of use cases and scale indefinitely, we wouldn't need to write an interoperability protocol. Maybe the "diversity of use cases" part could be clearer.

Reply:
> but I think it is derived from the primary goal of a scalable

Hmm, my point is that I think it deserves its own paragraph. 

> if a single ledger could handle a diversity of use cases and scale indefinitely, we wouldn't need to write an interoperability protocol

We would, e.g. for sovereignty. There is also the idea of optimizing each ledger for their use cases. 

As I said, if two ledgers do not scale, it is still interesting for them to interoperate. Even if it does not help with scalability. Also we have been pitching IBC for itneroeprability for the longest time. I do think it deserves a small paragraph.

Reply:
I agree with sovereignty as a separate goal.

Reply:
Added a paragraph - https://github.com/cosmos/ics/pull/70/commits/ba133b0ccd07870a3a185db7a8bc78554a5d0962

Thoughts?

Reply:
Spot on :+1: